bin/install

```diff
@@ -1,115 +1,137 @@
-    echo "Writing back-up copy → ${fish_config_bk}"
+  printf "$message\n"
+end
 
-    cp "${fish_config_file}" "${fish_config_bk}" >/dev/null 2>&1
-    test $? -ne 0 && die "Writing back-up copy failed, error code → ${?}"
+function fish_version_compatible
+  set -l MAJOR (echo $FISH_VERSION | cut -d. -f1)
+  set -l MINOR (echo $FISH_VERSION | cut -d. -f2)
+
+  return (test $MAJOR != OMF_FISH_MIN_VER[1] -o $MINOR -ge OMF_FISH_MIN_VER[2])
+end
+
+function install_omf
+  # Grant repository URL ends with .git
+  set git_uri (echo $OMF_REPO_URI | sed 's/\.git//').git
+
+  report progress "Cloning Oh My Fish $OMF_REPO_BRANCH from $git_uri..."
+  if not git clone -q --depth 1 -b $OMF_REPO_BRANCH $git_uri "$OMF_PATH"
+    report error "Error cloning Oh My Fish repository!"
+  end
+
+  set git_upstream (git --git-dir "$OMF_PATH/.git" --work-tree "$OMF_PATH" config remote.upstream.url)
+
+  if test -z "$git_upstream"
+    git --git-dir "$OMF_PATH/.git" --work-tree "$OMF_PATH" remote add upstream $git_uri
+  else
+    git --git-dir "$OMF_PATH/.git" --work-tree "$OMF_PATH" remote set-url upstream $git_uri
+  end
+
+  set fish_config_file "$FISH_CONFIG/config.fish"
+
+  if not test -e "$fish_config_file"
+    if mkdir -p (dirname "$fish_config_file")
+      touch "$fish_config_file"
+    end
+  end
+
+  if test -w "$fish_config_file"
+    report progress "Appending Oh My Fish startup to $fish_config_file..."
+
+    printf "\n# Setup Oh My Fish path and load it.\n" >> "$fish_config_file"
+    echo   "set -gx OMF_PATH $OMF_PATH"               >> "$fish_config_file"
+    if test "$OMF_CONFIG" != "$XDG_CONFIG_HOME/omf"
+      echo   "set -gx OMF_CONFIG $OMF_CONFIG"         >> "$fish_config_file"
+    end
+    echo   "source \$OMF_PATH/init.fish"              >> "$fish_config_file"
   else
-    omf_create_fish_config $fish_config_file
-  fi
-
-  echo "Adding Oh My Fish bootstrap → ${fish_config_file}"
-  touch ${fish_config_file} >/dev/null 2>&1
-  test ! -w ${fish_config_file} && die "Fish configuration file is not writable, aborting."
-
-  sed "s|{{OMF_PATH}}|$(echo "${OMF_PATH}" | sed -e "s|$HOME|\$HOME|")|" \
-    "${OMF_PATH}/templates/config.fish" > "${fish_config_file}"
-
-  if [ ! -d "${OMF_CONFIG}" ]; then
-    echo "Writing Oh My Fish configuration → ${OMF_CONFIG}"
-    mkdir -p "${OMF_CONFIG}"
-    test -f "${OMF_CONFIG}/bundle"    || echo "theme default" > "${OMF_CONFIG}/bundle"
-    test -f "${OMF_CONFIG}/theme"     || echo default > "${OMF_CONFIG}/theme"
-    test -f "${OMF_CONFIG}/revision"  || echo ${git_rev} > "${OMF_CONFIG}/revision"
+    report error "Aborting: Fish configuration file is not writable."
+  end
+
+  if not test -d "$OMF_CONFIG"
+    report progress "Creating Oh My Fish configuration..."
+    mkdir -p "$OMF_CONFIG"
+    test -f "$OMF_CONFIG/bundle"; or echo "theme default" > "$OMF_CONFIG/bundle"
+    test -f "$OMF_CONFIG/theme"; or echo "default" > "$OMF_CONFIG/theme"
+  end
+
+  source $fish_config_file
```

Comment:
Now that we are no longer overwriting the configuration file. We could crash installation here if the user has a statement on the `$fish_config_file` that has an `exit` call for instance.